File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/dtos/SearchRequest.java
package dev.jefy.connectpro.marketplace.applicaion.dtos;

import java.util.UUID;

import jakarta.validation.constraints.Min;

/**
 * @author Jôph E.F. Yamba
 * @date 08/04/2026
 * @time 12:33:01
 */
/**
 * @author Jôph Yamba
 */
public record SearchRequest(
        String search,
        @Min(1)
        int page,
        @Min(10)
        int size,
        UUID categoryId
) {}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/dtos/SearchRequest.kt
package dev.jefy.connectpro.marketplace.application.dtos

import jakarta.validation.constraints.Min
import java.util.*

/**
 * @author Jôph Yamba
 */

data class SearchRequest(
        val search: String?, 
        val page: Int = 0, 
        val size: Int = 10, 
        val categoryId: UUID?
)
